src/Features/LanguageServer/Protocol/AbstractRequestHandlerProvider.cs

```diff
@@ -19,9 +19,13 @@ namespace Microsoft.CodeAnalysis.LanguageServer
     internal abstract class AbstractRequestHandlerProvider
     {
         private readonly ImmutableDictionary<string, Lazy<IRequestHandler, IRequestHandlerMetadata>> _requestHandlers;
+        private readonly RequestExecutionQueue _queue;
 
-        public AbstractRequestHandlerProvider(IEnumerable<Lazy<IRequestHandler, IRequestHandlerMetadata>> requestHandlers, string? languageName = null)
-            => _requestHandlers = CreateMethodToHandlerMap(requestHandlers.Where(rh => rh.Metadata.LanguageName == languageName));
+        public AbstractRequestHandlerProvider(IEnumerable<Lazy<IRequestHandler, IRequestHandlerMetadata>> requestHandlers, RequestExecutionQueue queue, string? languageName = null)
```

Comment:
what does it mean for langaugename to be null?  we would then filter down to request handlers that don't answer requests for a particular langauge?  this seems.... odd.

i'm also not a fan of internal optiona parameters.  it just means it's hard to tell at the callsite what's going on.

Reply:
This was a new option added for XAML support.  The old handlers weren't updated when this option was added for XAML
Filed https://github.com/dotnet/roslyn/issues/46880 for updating this

Reply:
This is also covered by https://github.com/dotnet/roslyn/issues/46195 right? We need to split the handers into C# and VB forms for this too.

Reply:
right, I forgot about that one 😆 
Should be fixed once we switch to content type